### PR TITLE
Add _.findLastIndex and handle NaN in *IndexOf

### PR DESCRIPTION
**Pull request by megawac**
This fixes https://github.com/jashkenas/underscore/issues/1750 and adds a `_.findLastIndex` function

### CODE REVIEW

File: underscore.js [hunk 1]
```diff
@@ -648,9 +654,18 @@
   // Returns the first index on an array-like that passes a predicate test
   _.findIndex = function(array, predicate, context) {
     predicate = cb(predicate, context);
-    var length = array != null ? array.length : 0;
-    for (var i = 0; i < length; i++) {
-      if (predicate(array[i], i, array)) return i;
+    var length = array != null && array.length;
+    for (var index = 0; index < length; index++) {
```
Comment (jridgewell) [inline underscore.js]: Is there a reason to use `index` over `i`?

Comment (megawac) [inline underscore.js]: Nope, but we should make up our mind and be consistent.

Comment (jridgewell) [inline underscore.js]: I think it's already `i`. :stuck_out_tongue:

Comment (megawac) [inline underscore.js]: Thanks for the :eyes: @jridgewell

**Comment by megawac** [comment]
Any objections to this current implementation?

**Pull request merged**